src/tensors/tensor.jl

```diff
@@ -365,50 +266,71 @@ Construct a `TensorMap` from a plain multidimensional array.
   `S<:ElementarySpace`.
 - `tol=sqrt(eps(real(float(eltype(data)))))::Float64`: 
     
-Here, `data` can be specified in two ways. It can either be a `DenseArray` of rank `N₁ + N₂`
-whose size matches that of the domain and codomain spaces,
-`size(data) == (dims(codomain)..., dims(domain)...)`, or a `DenseMatrix` where
-`size(data) == (dim(codomain), dim(domain))`. The `TensorMap` constructor will then
-reconstruct the tensor data such that the resulting tensor `t` satisfies
-`data == convert(Array, t)`. For the case where `sectortype(S) == Trivial`, the `data` array
-is simply reshaped into matrix form and referred to as such in the resulting `TensorMap`
-instance. When `S<:GradedSpace`, the `data` array has to be compatible with how how each
-sector in every space `V` is assigned to an index range within `1:dim(V)`.
-
-Alternatively, the domain and codomain can be specified by passing a [`HomSpace`](@ref)
-using the syntax `codomain ← domain` or `domain → codomain`.
+Here, `data` can be specified in three ways:
+1) `data` can be a `DenseVector` of length `dim(codomain ← domain)`; in that case it represents
+   the actual independent entries of the tensor map. An instance will be created that directly
+   references `data`.
+2) `data` can be an `AbstractMatrix` of size `(dim(codomain), dim(domain))`
+3) `data` can be an `AbstractArray` of rank `N₁ + N₂` with a size matching that of the domain
+   and codomain spaces, i.e. `size(data) == (dims(codomain)..., dims(domain)...)`
+
+In case 2 and 3, the `TensorMap` constructor will reconstruct the tensor data such that the
+resulting tensor `t` satisfies `data == convert(Array, t)`, up to an error specified by `tol`.
+For the case where `sectortype(S) == Trivial` and `data isa DenseArray`, the `data` array is
+simply reshaped into a vector and used as in case 1 so that the memory will still be shared.
+In other cases, new memory will be allocated.
+
+Note that in the case of `N₁ + N₂ = 1`, case 3 also amounts to `data` being a vector, whereas
+when `N₁ + N₂ == 2`, case 2 and case 3 both require `data` to be a matrix. Such ambiguous cases
+are resolved by checking the size of `data` in an attempt to capture all support all possible
```

Comment:
```suggestion
are resolved by checking the size of `data` in an attempt to support all possible
```

Reply:
I have fixed this locally and will push it in a next commit, as committing this here is more annoying to then not have conflicts with other changes I already have locally.